Finalized pages, requesting peer review

jacobszmon opened the pull request:
I finalized my initial and editing passes on my pages.

Requesting peer review before merging with main.

jacobszmon committed:
Initialized my branch (jake) and added pages.

Added git-fundamentals.md and its subpages with template formatting.
- checkout.md: will contain text explaining the checkout function.
- reset.md: will contain text explaining the reset function.
- revert.md: will contain text explaining the revert function.
- clean.md: will contain text explaining the clean function.
- when-to-use-different-strategies.md : will contain text explaining the which of the above functions to use in various scenarios.

jacobszmon committed:
Correcting mistakes and adding remaining base files.

- Corrected mistakes in the base of undoing-in-git-with.md

- Added base templates for:
  - resolving-resolving-merge-conflict.md
  - stashing.md

jacobszmon committed:
Corrected odd formatting issues.

When I made the last commit, some strange formatting issues arose and I have no clue where they came from. The problems were simple to fix, so I fixed them.

jacobszmon committed:
Filled out five pages.

I added automatic table of contents sections to five files, then filled out their required contents. The files affected are as follows:
- checkout.md
- reset.md
- revert.md
- clean.md
- when-to-use-different-strategies.md

I also enabled "has_children" on the parent page for these files.

jacobszmon committed:
Added base contents to "undoing-in-git-with.md"

I added the base needed contents to the file "undoing-in-git-with.md"

jacobszmon committed:
Added Base Contents to "stashing.md"

jacobszmon committed:
Added Base Contents to "resolving-merge-conflicts.md"

jacobszmon committed:
Added/Updated Finished Task List.

Accidentally cloned a version of the repo without a task list in the README.md, so I grabbed the task list from the main branch and added/updated it with my contributions here.

jacobszmon committed:
Added sources for Information on each page.

I went through all my pages and added a section where I sourced my information.
I nested the links in text explaining what the source is so a reader doesn't need to click to know where I got it from.

Samuel-The-Daya commented:
This is all good, but if wanted to, even though all the information is copy and paste from slides, we do have to create a link of resources about our topics, its in fine print for the assignment, but it is an easy task.

jacobszmon committed:
Changed File Names for Easier Navigation

Added new folders for each base page and renamed all pages according to these guidelines:
- Folder numbers denote the internal "nav_order" of the page.
- Pages without children are numbered to match their "nav_order".
- Pages with children are numbered "00" to ensure they appear at the top of their respective folder.
- Child pages are numbered according to their local "nav_order"

jacobszmon committed:
Updated To-Do list in README.md

I updated the To-Do list in README.md to make the task distribution more readable and condensed.
Then I added two new tasks for me to finish.
